modules/features2d/src/opencl/brute_force_match.cl

```diff
@@ -400,6 +398,7 @@ __kernel void BruteForceMatch_knnMatch(
     query_cols /= kercn;
 
     __local float sharebuffer[SHARED_MEM_SZ];
+    __local float sharebuffer2[SHARED_MEM_SZ];
```

Comment:
I think this is not good idea to use additional local memory. We will have performance degradation on some devices.